superset/commands/importers/v1/utils.py

```diff
@@ -214,3 +218,80 @@ def get_contents_from_bundle(bundle: ZipFile) -> dict[str, str]:
         for file_name in bundle.namelist()
         if is_valid_config(file_name)
     }
+
+
+# pylint: disable=consider-using-transaction
+def import_tag(
+    new_tag_names: list[str],
+    contents: dict[str, Any],
+    object_id: int,
+    object_type: str,
+    db_session: Session,
+) -> list[int]:
+    """Handles the import logic for tags for charts and dashboards"""
+
+    if not feature_flag_manager.is_feature_enabled("TAGGING_SYSTEM"):
+        return []
+
+    tag_descriptions = {}
+    new_tag_ids = []
+    if "tags.yaml" in contents:
+        try:
+            tags_config = yaml.safe_load(contents["tags.yaml"])
+        except yaml.YAMLError as err:
+            logger.error("Error parsing tags.yaml: %s", err)
+            tags_config = {}
+
+        for tag_info in tags_config.get("tags", []):
+            tag_name = tag_info.get("tag_name")
+            description = tag_info.get("description", None)
+            if tag_name:
+                tag_descriptions[tag_name] = description
+    existing_tags = (
+        db_session.query(TaggedObject)
+        .filter_by(object_id=object_id, object_type=object_type)
+        .all()
+    )
+
+    for tag_name in new_tag_names:
+        try:
+            tag = db_session.query(Tag).filter_by(name=tag_name).first()
```

Comment:
This is going to be potentially super slow, better to move the query outside of the loop. Something like:

```python
existing_tags = {tag.name: tag for tag in db_session.query(Tag).filter(tag.name.in_(new_tag_names))}
```

Then you can do:

```python
for tag_name in new_tag_names:
    tag = existing_tags.get(tag_name)
    if tag is None:
        tag = Tag(...)
```